### PR TITLE
[ADD] Email templates for closed tickets and changed state

### PR DESCRIPTION
**Pull request by miguel-domatix**
Email templates to use when the ticket reaches a stage that has the associated template.

### CODE REVIEW

File: helpdesk_mgmt/data/helpdesk_data.xml [hunk 1]
```diff
@@ -22,6 +22,169 @@
...
+                                    </a>
+                                % endif
+                              </td>
+                            </tr>
+                          </table>
+                        </td>
+                      </tr>
+                      <tr>
+                        <td align="center">
+                            Powered by <a target="_blank" href="https://www.odoo.com">Odoo</a>.
```
Comment (pedrobaeza) [inline helpdesk_mgmt/data/helpdesk_data.xml]: Why adding this text?

File: helpdesk_mgmt/data/helpdesk_data.xml [hunk 2]
```diff
@@ -22,6 +22,169 @@
...
```
Comment (pedrobaeza) [inline helpdesk_mgmt/data/helpdesk_data.xml]: Why adding this text?

**Review by pedrobaeza — COMMENTED** [review]
Not sure that the stage change should be notified to the customer. And how is this sent? This is only adding the templates, not any automation code.

**Comment by miguel-domatix** [comment]
This PR is only for template emails. Furthermore, each stage can have a email template that notify. Then, if in this PR must be the actions for templates, tell me.
Thank you.

**Comment by pedrobaeza** [comment]
Well, I think the added value is to have them usable installing the module without extra steps.

**Comment by miguel-domatix** [comment]
Do I have to associate the template with the corresponding states from the data folder?

**Comment by pedrobaeza** [comment]
I don't know right now how do you get the mail sent. Is there such automation for auto-sending a mail template when the stage is changed like the one in tasks? If so, then linking it with the stage in data is enough.

**Comment by miguel-domatix** [comment]
In helpdesk_ticket.py file, there are a function that send a email if the state changes if it has an associated template. The function's name is _track_template.

**Comment by pedrobaeza** [comment]
Perfect, then just link that template to the standard stages populated in data

**Comment by pedrobaeza** [comment]
Any way, my doubt about spamming user each time you change a stage is there.

**Comment by miguel-domatix** [comment]
In the states form you can configure it and associate or not any email template.

**Comment by pedrobaeza** [comment]
Yeah, then it's better to not link that template by default.

**Comment by miguel-domatix** [comment]
Fine, then I let it like this.

**Comment by pedrobaeza** [comment]
But this is not linking the closed template... What I don't think it's good is to link the stage changed template, but ticket closed template should be.

**Comment by miguel-domatix** [comment]
The email templates aren't shown on chatter, but it works on mail. Do I make a simpler template for chatter and these two for mail?

**Comment by pedrobaeza** [comment]
Please look at project stages, where you have a mail template linked to them, and when you changed the stage, there's an automation that sends that template. That's the idea here, with the closed stage (provided as data with that template linked).

**Comment by miguel-domatix** [comment]
Ok. Then, this templates is not seen in chatter. I think the chatter can't render the template. That may be?.

**Comment by pedrobaeza** [comment]
Chatter shows any message sent in the thread, with or without template, with or without "advanced formatting". Please review task stages code for knowing how to send a message with a template via code.

**Comment by miguel-domatix** [comment]
Did you mean do it like that?

**Comment by pedrobaeza** [comment]
@miguel-domatix yes, finally you got me!

**Comment by miguel-domatix** [comment]
Fine!

**Review by alvaro-domatix — APPROVED** [review]

**Review by samuel-domatix — APPROVED** [review]

**Pull request closed without merging**